app/views/CannedResponsesListView/index.js

```diff
@@ -0,0 +1,243 @@
+
+const COUNT = 25;
+
+const fixedScopes = [
+	{
+		_id: 'all',
+		name: I18n.t('All')
+	},
+	{
+		_id: 'public',
+		name: I18n.t('Public')
+	},
+	{
+		_id: 'user',
+		name: I18n.t('Private')
+	}
+];
+
+const CannedResponsesListView = ({ navigation, route }) => {
+	const [cannedResponses, setCannedResponses] = useState([]);
+	const [cannedResponsesScopeName, setCannedResponsesScopeName] = useState([]);
+	const [searchText, setSearchText] = useState('');
+	const [scope, setScope] = useState('');
+	const [departments, setDepartments] = useState([]);
+	const [allDepartments, setAllDepartments] = useState([]);
+	const [departmentId, setDepartmentId] = useState('');
+	const [loading, setLoading] = useState(true);
+	const [offset, setOffset] = useState(0);
+
+	const { room } = route.params;
+
+	const { theme } = useTheme();
+	const { isMasterDetail } = useSelector(state => state.app);
+	const { rooms } = useSelector(state => state.room);
+
+	useEffect(() => {
+		navigation.setOptions({
+			title: I18n.t('Canned_Responses')
+		});
+	}, []);
+
+	const getDepartments = debounce(async(keyword = '') => {
+		try {
+			const res = await RocketChat.getDepartments(keyword);
+			const regExp = new RegExp(keyword, 'gi');
+			const filterWithText = fixedScopes.filter(dep => regExp.test(dep.name));
+			res.success
+				? setDepartments([...filterWithText, ...res.departments])
+				: setDepartments(filterWithText);
+
+			if (res.success && !keyword) { setAllDepartments([...filterWithText, ...res.departments]); }
+		} catch {
+			// do nothing
+		}
+	}, 300);
+
+	const goToDetail = (item) => {
+		navigation.navigate('CannedResponseDetail', { cannedResponse: item, room });
+	};
+
+	const navigateToRoom = (item) => {
+		const { name, username } = room;
+		const params = {
+			rid: room.rid,
+			name: RocketChat.getRoomTitle({
+				t: room.t,
+				fname: name,
+				name: username
+			}),
+			t: room.t,
+			roomUserId: RocketChat.getUidDirectMessage(room),
+			usedCannedResponse: item.text
+		};
+
+		if (room.rid) {
+			// if it's on master detail layout, we close the modal and replace RoomView
+			if (isMasterDetail) {
+				Navigation.navigate('DrawerNavigator');
+				goRoom({ item: params, isMasterDetail, usedCannedResponse: item.text });
+			} else {
+				let navigate = navigation.push;
+				// if this is a room focused
+				if (rooms.includes(room.rid)) {
+					({ navigate } = navigation);
+				}
+				navigate('RoomView', params);
+			}
+		}
+	};
+
+	const getListCannedResponse = async(text, department, depId, debounced) => {
+		try {
+			const res = await RocketChat.getListCannedResponse({
+				text,
+				offset,
+				count: COUNT,
+				departmentId: depId,
+				scope: department
+			});
+			if (res.success) {
+				setCannedResponses(prevCanned => (
+					debounced
+						? res.cannedResponses
+						: [...prevCanned, ...res.cannedResponses]
+				));
+				setLoading(false);
+				setOffset(prevOffset => prevOffset + COUNT);
```

Comment:
Offset is not something you're going to render or that impacts directly on render, so it should not be a state value.
It could be a `let` imo